drivers/dma/dma_stm32.c

```diff
@@ -93,6 +109,11 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 
 	stream = &config->streams[id];
 
+	/* Do not continue if dma channel is not busy anymore */
+	if (stream->busy == false) {
+		return;
+	}
+
```

Comment:
If affecting existing DMAs, please move it to a dedicated PR.
(In case of revert is required for instance, but also to allow users that are interested in non v3 dmas to review)